kyuubi-main/src/main/scala/org/apache/kyuubi/engine/SQLEngineAppName.scala

```diff
@@ -32,6 +32,13 @@ case class SQLEngineAppName(
       case _ => ZKPaths.makePath(s"${prefix}_$sharedLevel", user)
     }
   }
+
+  def getZkLockPath(prefix: String): String = {
+    assert(sharedLevel != CONNECTION)
+    sharedLevel match {
+      case _ => ZKPaths.makePath(s"${prefix}_$sharedLevel", "lock", user)
```

Comment:
`/kyuubi/kyuubi_USER/lock/userName/lock/lockes`, is the first  above `lock` neccessary?

Reply:
Yes. `GetServiceNodesInfo` cannot touch the `lock` node.
https://github.com/NetEase/kyuubi/blob/master/kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala#L266